docs/usage.rst

```diff
@@ -50,53 +40,39 @@ Cut-and-paste into your config file:
-``.ini`` file.  For example:
+^^^^^^^^^^^^^^^^^^
+If you have included :mod:`pyramid_nacl_session` in your Pyramid project's configuration as shown above then :func:`pyramid_nacl_session.session_factory_from_settings` is called automatically and you need do nothing else.
+Otherwise you will need to create a Pyramid :term:`session factory` by adding a call to either the :func:`pyramid_nacl_session.EncryptedCookieSessionFactory` function or the :func:`pyramid_nacl_session.session_factory_from_settings` function in the configuration code of your Pyramid project's ``__init__.py`` file and subsequently register that session factory with Pyramid.
+At that point, accessing ``request.session`` will provide a Pyramid session using PyNaCl as a backend.
+
+:func:`pyramid_nacl_session.session_factory_from_settings` obtains session settings from the ``**settings`` dictionary passed to the Configurator.
+It assumes that you've placed session configuration parameters prefixed with ``session.`` in your Pyramid application's ``.ini`` file.  
+
+For example:
 
 .. code-block:: ini
 
    [app:myapp]
-   .. other settings ..
+   # other settings 
    session.secret = 840aaafdc36f067fbad9baf006efc0f672b86ab0dcb6a3e43ecc1f9d760915e5
    session.serializer = json
 
-If your ``.ini`` file has such settings, you can use
-:func:`pyramid_nacl_session.session_factory_from_settings` in your
-application's configuration.  For example, let's assume this code is
-in the ``__init__.py`` of your Pyramid application that uses an
-``.ini`` file with the ``session.`` settings above to obtain its
-``**settings`` dictionary.
+If your ``.ini`` file has such settings, you can use :func:`pyramid_nacl_session.session_factory_from_settings` in your application's configuration.
+For example, let's assume this code is in the ``__init__.py`` of your Pyramid application that uses an ``.ini`` file with the ``session.`` settings above to obtain its ``**settings`` dictionary.
 
 .. code-block:: python
 
    from pyramid_nacl_session import session_factory_from_settings
    from pyramid.config import Configurator
 
    def app(global_config, **settings):
-       """ This function returns a WSGI application.
+      """ This function returns a WSGI application.
        
-       It is usually called by the PasteDeploy framework during 
-       ``paster serve``.
-       """
-       session_factory = session_factory_from_settings(settings)
-       config = Configurator(root_factory=get_root, settings=settings)
-       config.set_session_factory(session_factory)
-       # ... other configuration stuff...
-       return config.make_wsgi_app()
+      It is usually called by the PasteDeploy framework during 
+      ``paster serve``.
+      """
+      session_factory = session_factory_from_settings(settings)
+      with Configurator(settings=settings) as config:
+         config.set_session_factory(session_factory)
+         # other configuration stuff
+         return config.make_wsgi_app()
```

Comment:
All the above lines 65-78 are not correctly indented. Please use 4 spaces for reStructuredText and Python code samples.

Also please check other indentation earlier in this file to ensure it is 4 spaces.

Note, PyCharm is notorious for inheriting the first indentation as its default for the document which will cause Python code samples to be incorrectly spaced. Keeping all indents to 4 spaces avoids that issue. Other editors might exhibit the same behavior.

Reply:
> Other editors might exhibit the same behavior.

indeed, same in Neovim :smiling_imp: